### PR TITLE
Add new actions for IP address valuation to Linux Pack.

### PR DESCRIPTION
**Pull request by jjm**
Complete!

- Add ipaddress module to requirements.txt in linux pack.
- Add new actions is_valid_ip, is_valid_ipv4 and is_valid_ipv6
  to linux pack.
- Add tests for is_valid_ip.py action script.
- Add entry to for new actions to the CHANGELOG.rst.

### CODE REVIEW

File: contrib/linux/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 # used by file watcher sensor
 pyinotify>=0.9.5,<=0.10
 -e git+https://github.com/Kami/logshipper.git@stackstorm_patched#egg=logshipper
+ipaddress>=1.0.16
```
Comment (Kami) [inline contrib/linux/requirements.txt]: IIRC (but I need to double check how new packages handle that) we need to also put this in global requirements.txt file.

Comment (Kami) [inline contrib/linux/requirements.txt]: (Just thinking out loud) - Module license is Python Software Foundation License v2 (https://github.com/phihag/ipaddress/blob/master/LICENSE) so it's OK for us to bundle this dependency.

File: contrib/linux/actions/is_valid_ip.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import ipaddress
+
+from st2actions.runners.pythonrunner import Action
+
+
+class IsValidIp(Action):
```
Comment (Kami) [inline contrib/linux/actions/is_valid_ip.py]: Minor style thing - we usually suffix action classes with `Action`.

**Comment by Kami** [comment]
Thanks.

The changes look good to me :+1:

**Comment by Kami** [comment]
Please also sync this branch with latest master since there are some conflicts it looks like.

**Comment by jjm** [comment]
Sorry, I seem to have sync'ed our master branch and then branched from the wrong place :-( I'll fix.

**Comment by Kami** [comment]
Since this package is a port of Python 3 functionality, I believe it should also work on Windows.

In that case, `core` pack is maybe a better place for this action. What do others think (in any case, it's not such a big deal right now since we don't really officially support running StackStorm on Windows anyway)?

/cc @lakshmi-kannan @manasdk

**Comment by jjm** [comment]
I've corrected the Action suffix and brought my branch up to date, so should be fine to merge.

**Comment by Kami** [comment]
Thanks.

I'll wait for the feedback on the pack and merge it after that :)

File: contrib/linux/requirements.txt (hunk 1 above)
Comment (lakshmi-kannan) [inline contrib/linux/requirements.txt]: >IIRC (but I need to double check how new packages handle that) we need to also put this in global requirements.txt file.

Yep. That or move linux pack back to st2contrib.

**Comment by lakshmi-kannan** [comment]
I like this in linux pack just fine but I have problems with linux pack being part of st2 code. I lost that argument before but I am not giving up.

**Comment by lakshmi-kannan** [comment]
It appears no one has a problem moving linux pack back to st2contrib. Once this PR gets merged, I'll own moving it to st2contrib.

**Comment by manasdk** [comment]
I am not very clear what these actions have to do anything with linux or with core StackStorm functionality. Would it make more sense to place them in st2contrib in a standalone pack called `ip_utils`?

**Comment by jjm** [comment]
They are general utility / helper functions for use in other workflows,so following a quick discussion on Slack it was thought the linux pack in ST2 would be best location.

I think the real question is _Where do you put general propose helper actions?_ They are of use only in other workflows, if they are not the main st2 repo will folks be allowed to use them in other st2contrib workflows? As the normal review comment is often _don't depend on other packs_ if you use action from another pack.

My suggestion is move them to a new *utils* pack in st2, which purpose is useful helper actions for workflows. We can then add actions like the following:

- is_valid_hostname
- is_valid_dns_entry
- is_in_dns

Which would all be every useful for adding checks within workflows, I guess the other option wold be aiming to push them into Mistral? Which would reduce the load on the action runners..

These came from my need to add a check for a valid IP in one of our troubleshooting flows and I thought it made sense to share with the wider community as it's action I see others will need to do, so why not same them the effort.

**Comment by Kami** [comment]
I'm personally fine with either moving it to a separate pack (e.g. `networking` / `ip_utils`) or leaving it in a `linux` pack.

We already have some "similar" functionality in Linux pack (e.g. wait_for_ssh) so I don't see it as a big problem.

**Comment by Kami** [comment]
> I like this in linux pack just fine but I have problems with linux pack being part of st2 code. I lost that argument before but I am not giving up.

This also means updating installer script and other stuff to install this pack by default? Or?

**Comment by manasdk** [comment]
> We already have some "similar" functionality in Linux pack (e.g. wait_for_ssh) so I don't see it as a big problem.

I guess we all draw our personal lines very differently. What is the similarity between `wait_for_ssh` and `is_valid_ip ` etc? The fact that it has something to do with ip addresses or is related to networking?

**Comment by Kami** [comment]
> I guess we all draw our personal lines very differently. What is the similarity between wait_for_ssh and is_valid_ip etc? The fact that it has something to do with ip addresses or is related to networking?

Both.

Anyway, to unblock this PR, is everyone OK with moving this to new "networking" or similar pack in st2contrib?

**Comment by jjm** [comment]
If you let me know where it would go, I'll open a new PR.

**Comment by jjm** [comment]
How about *networking_utils* in st2contrib?

**Comment by manasdk** [comment]
@jjim sounds good! New networking_utils it is

**Comment by jjm** [comment]
I've moved this to StackStorm/st2contrib/#498.

**Comment by jjm** [comment]
I've moved these actions to the networking-utils pack in StackStorm/st2contrib#498.

**Pull request closed without merging**